src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -14,55 +14,75 @@
 
 package com.google.sps.servlets;
 
-import data.User;
+import com.google.sps.data.User;
 import java.io.IOException;
 import com.google.gson.Gson;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 
-/** Servlet to store and get retrieve information. */
+/** Servlet to store and get retrieve user information. */
 @WebServlet("/user")
 public final class UserServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the id from the query string.
-    long id = request.getParameter("id");
+    UserService userService = UserServiceFactory.getUserService();
 
+    // Get the key from the query string.
+    String keyString = request.getParameter("key");
+    Key userKey = KeyFactory.stringToKey(keyString);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Entity userEntity = datastore.get(id);
-    String email = userEntity.getProperty("email");
-    String username = userEntity.getProperty("username");
-    User user = new User(id, email, username);
 
-    // Convert to JSON and send it as the response.
-    Gson gson = new Gson();
+    try{
+      Entity userEntity = datastore.get(userKey);
+      String email = (String) userEntity.getProperty("email");
+      String username = (String) userEntity.getProperty("username");
+      User user = new User(userKey.getName(), email, username);
 
-    response.setContentType("application/json");
-    response.getWriter().println(gson.toJson(user));
+      // Convert to JSON and send it as the response.
+      Gson gson = new Gson();
+
+      response.setContentType("application/json");
+      response.getWriter().println(gson.toJson(user));
+    }
+    catch(EntityNotFoundException e) {
+      throw new IOException("Entity not found.");
```

Comment:
How are you planning to use this exception?

Reply:
I have no plan, but did you have a suggestion?

Reply:
I don't know much about servlets, but I am thinking that such an exception should be reflected to the client side, either by having an error page, or a simple message with user not found/incorrect credentials, etc.